docs/quarkslab-audit/README.md

```diff
@@ -0,0 +1,60 @@
+
+The full report will be made available in this repository once it is released.
+
+The report did not identify any particular security issues, but provided recommendations to improve the codebase and make it more robust against future changes.
+An initial draft version of the report identified three findings: `LOW 1`, `INFO 1`, and `INFO 2`.
+We provide context and responses to each finding here, but direct readers to the full report for details.
+
+## `LOW 1`
+
+This issue noted that the [`merlin`](https://crates.io/crates/merlin) crate used by the implementation for Fiat-Shamir transcripting appears to be unmaintained, as it had not been updated in over two years at the time of writing.
+The report recommended that Tari produce a fork of this repository.
+
+### Response
+
+To our knowledge, no security-related issues have been reported that would render the use of the `merlin` crate unsafe.
+Further, no particular dependency conflicts have been identified due to the lack of apparent maintenance.
+
+For these reasons, producing a fork does not appear to be necessary, as this would merely increase complexity and technical debt.
+As with any core dependency, any future security-related issues arising in `merlin` would be addressed as appropriate.
+
+## `INFO 1`
+
+This issue noted the curve library [fork](https://crates.io/crates/tari-curve25519-dalek) used in the implementation is not up to date with the [upstream](https://crates.io/crates/curve25519-dalek) repository.
+The report recommended that Tari bring the fork up to date.
+
+### Response
+
+The fork was initially made due to the upstream repository being apparently unmaintained, which resulted in a number of dependency conflicts.
+Additionally, the fork added support for multiscalar multiplication [partial precomputation](https://github.com/tari-project/curve25519-dalek/pull/1) that was not supported by upstream.
+Since that time, upstream has undergone a flurry of activity and updates; during this process, the fork was rebased against the upstream [`4.0.0-rc.3`](https://github.com/dalek-cryptography/curve25519-dalek/releases/tag/4.0.0-rc.3) tag.
+Subsequent to this rebase, the upstream repository was significantly restructured, and the fork has not been updated to reflect this.
+We did not identify any particular security-related issues addressed since this time.
+Unfortunately, the versioning scheme used in the fork conflicts somewhat with the upstream versioning; the fork currently is on the [`v4.0.3`](https://github.com/tari-project/curve25519-dalek/releases/tag/v4.0.3) tag.
+
+The fork may be updated in the future against a more recent upstream tag if doing so does not introduce conflicts in Tari repositories.
+Further, an open [upstream pull request](https://github.com/dalek-cryptography/curve25519-dalek/pull/546) would add partial precomputation support; if it is accepted, the implementation may change its curve library dependency to upstream.
```

Comment:
I think we should update to their (`curve25519-dalek`) latest version and apply our specific changes to it in our fork, similar to what we did in the past.

Reply:
Certainly can. Upstream switched to a monorepo structure after the release candidates, but I doubt this would cause issues other than pulling in a lot of functionality that wouldn't be needed. I'll test this out.

Reply:
See [this issue](https://github.com/tari-project/bulletproofs-plus/issues/93).

Reply:
Addressed in the curve library fork by [this PR](https://github.com/tari-project/curve25519-dalek/pull/6).

Reply:
Unfortunately `tari-crypto` would be blocked on such an update by a dependency conflict. Might be best to stick with the current library version for now.